### ISSUE TITLE
Daily strips not displaying in widget on nexus 5 updated to Android 6.0

### ISSUE THREAD

**Issue by palmheads** [smarek/Simple-Dilbert#25]
Hi 

Curious issue - 4.2 Simple Dilbert on nexus 5 recently updated to Android 6.0, the weekend strip displays fine in widget, but every other day no strip displays.

In the app itself all strips appear fine. Just the widget where it struggles.

Cheers
Daryn

**Comment by smarek** [comment]
Hi, could you please take a peek in the LogCat if you see any problems there? Also what is your setting for the Widget, do you use it to display "latest only" or do you have it with ability to flip through days?
Have you tried tapping it (or the refresh button in the corner) to refresh the strip which doesn't display?
Thanks for report

**Comment by palmheads** [comment]
Hi Marek

Installed adb on grepped "dilbert" on my Nexus 5

10-27 18:22:12.330   778   791 I ActivityManager: Start proc
26205:com.mareksebera.simpledilbert/u0a255 for broadcast
com.mareksebera.simpledilbert/.widget.WidgetProvider
10-27 18:25:32.056   778   789 I ActivityManager: Process
com.mareksebera.simpledilbert (pid 26205) has died
10-27 18:41:07.821   778   791 I ActivityManager: Start proc
3037:com.mareksebera.simpledilbert/u0a255 for broadcast
com.mareksebera.simpledilbert/.widget.WidgetProvider
10-27 18:49:24.173   778  3580 I ActivityManager: Killing
3037:com.mareksebera.simpledilbert/u0a255 (adj 15): empty #17
10-27 18:52:23.473   778   791 I ActivityManager: Start proc
5078:com.mareksebera.simpledilbert/u0a255 for broadcast
com.mareksebera.simpledilbert/.widget.WidgetProvider
10-27 18:54:50.284   778  3582 I ActivityManager: Killing
5078:com.mareksebera.simpledilbert/u0a255 (adj 15): empty #17
10-27 18:57:10.203   778   791 I ActivityManager: Start proc
6985:com.mareksebera.simpledilbert/u0a255 for broadcast
com.mareksebera.simpledilbert/.widget.WidgetProvider
10-27 18:59:05.904   778  1855 I ActivityManager: Process
com.mareksebera.simpledilbert (pid 6985) has died
10-27 18:59:19.276   778  3582 I ActivityManager: Force stopping
com.mareksebera.simpledilbert appid=10255 user=0: from pid 5586
10-27 18:59:19.277   778  3582 I ActivityManager:   Force finishing
activity ActivityRecord{3eeeb2d u0
com.mareksebera.simpledilbert/.core.DilbertFragmentActivity t15934}
10-27 19:00:26.504  5586  5586 I AppStorageSettings: Clearing user data for
package : com.mareksebera.simpledilbert
10-27 19:00:26.505   778  2209 I ActivityManager: Force stopping
com.mareksebera.simpledilbert appid=10255 user=0: clear data
10-27 19:00:26.617  5586  5586 I AppStorageSettings: Cleared user data for
package : com.mareksebera.simpledilbert
10-27 19:00:26.699 13813  9666 D PackageBroadcastService: Received
broadcast action=android.intent.action.PACKAGE_DATA_CLEARED and
uri=com.mareksebera.simpledilbert
10-27 19:00:26.718 13813  9666 D AccountUtils: Clearing selected account
for com.mareksebera.simpledilbert
10-27 19:00:26.736 13813 14646 I Icing   : doRemovePackageData
com.mareksebera.simpledilbert
10-27 19:00:26.773 13813  9666 I LocationSettingsChecker: Removing dialog
suppression flag for package com.mareksebera.simpledilbert
10-27 19:00:37.441   778   791 I ActivityManager: Start proc
9927:com.mareksebera.simpledilbert/u0a255 for broadcast
com.mareksebera.simpledilbert/.widget.WidgetProvider
10-27 19:00:37.510  9927  9927 W System  : ClassLoader referenced unknown
path: /data/app/com.mareksebera.simpledilbert-1/lib/arm



Basically I refreshed the widget a couple of times, went into Apps &
cleared cache & data etc just to make sure that was clear. Still not
working on a smaller mon->sat cartoon. Toggled only show current etc. Got
nothing

Is that last line in log the issue?
10-27 19:00:37.510  9927  9927 W System  : ClassLoader referenced unknown
path: /data/app/com.mareksebera.simpledilbert-1/lib/arm


cheers
Daryn

**Comment by CraigsBar** [comment]
I have the same issue on cm13 (os 6.0.1) on my HTC one m8.

**Comment by mvrk80** [comment]
I have the same issue on Lg g4 after the Android 6 official Lg update :(

**Comment by supergonzales** [comment]
Same issue on a Nexus 5X on Android 6.0.1, only the Sunday strips can be seen. The other strips seem to be downloaded as the loading animation is showing but then the widget remains blank. Refreshing only repeats this process.

**Comment by mvrk80** [comment]
please please please
we need a new version with this bug fixed :_(

**Comment by sorab2142** [comment]
07-01 13:50:27.556: E/OpenGLRenderer(16823): Cannot generate texture from bitmap

gets printed to logcat, when creating the widget and also when clicking any of the buttons (random,refresh,next,previous).

**Comment by smarek** [comment]
Ok, investigated and fixed. For those who are interested in problem:

**E/OpenGLRenderer: Cannot generate texture from bitmap** happens on Android 6+ when you try to load GIF image through `setImageViewBitmap` on `RemoteViews`

Also, Sunday strips are OK because those are served as JPEG (JFIF), daily strips are served as GIFs (dunno why)

Possible solutions are:
1) Migrate from GIF to PNG/JPG/...  (not doable, amuniversal.com serves only GIFs for daily strips, and PNG for Sunday strips, ie. see here: http://assets.amuniversal.com/ddb9490016c20134707c005056a9545d )
2) Download the GIF, encode it as PNG, then decode it from file and push the Bitmap through `setImageViewBitmap`, this (and other problems) are solved by using Glide library

**Comment by smarek** [comment]
Version v4.3 is tagged so please let F-Droid team some time to publish new build, thank you

**Comment by mvrk80** [comment]
thank you smarek!!
long to download the new version

**Comment by mvrk80** [comment]
not available yet on f-droid :(

**Comment by smarek** [comment]
I can see the version being pushed to FDroid Data repo, on 25th Jun, don't know, why it's not available in latest data snapshot yet, see
https://gitlab.com/fdroid/fdroiddata/commit/3a07cc998e5233302c5913c7144cf18a1c462915
https://gitlab.com/fdroid/fdroiddata/commit/2701ae0a8b0b2128a0dcdd437be39a0016df0634

**Comment by mvrk80** [comment]
still having similar issue on 3.4 : the widget now show the strip only if I open it in the app before.
And the share function is not working anymore🙁
